pkg/api/testing/defaulting_test.go

```diff
@@ -225,45 +225,56 @@ func TestDefaulting(t *testing.T) {
 	sort.Sort(testTypes)
 
 	for _, gvk := range testTypes {
-		_, expectedChanged := typesWithDefaulting[gvk]
-		iter := 0
-		changedOnce := false
-		for {
-			if iter > *roundtrip.FuzzIters {
-				if !expectedChanged || changedOnce {
-					break
+		t.Run(gvk.String(), func(t *testing.T) {
+			_, expectedChanged := typesWithDefaulting[gvk]
+			iter := 0
+			changedOnce := false
+			for {
+				if iter > *roundtrip.FuzzIters {
+					if !expectedChanged || changedOnce {
+						break
+					}
+					// This uses to be 300, but for ResourceClaimList that was not high enough
```

Comment:
No, it's a few seconds before and after. Each individual gvk doesn't take long, it just adds up.

Reply:
It's reallyunclear to me what this test is achieveing.  AFAICT, all it does is prove that 1 field got a default value, 1 time.  It does not prove that it was CORRECT or that it got ALL of them, just that it changed one thing one time.  And as this PR shows it is non-deterministic.

Why not set this to 10K or 100K or 1M?

Reply:
> It's really unclear to me what this test is achieveing. [...]

I think your analysis is correct. I suppose it was written because a generic validation test is still better than no test at all, but I am just guessing.

> Why not set this to 10K or 100K or 1M?

If it's too large, then a gvk with no defaulting might really run too long.